### PR TITLE
Character Warnings ext to core

### PR DESCRIPTION
**Pull request by SpeedyD**
Pulled @redcr0wz 's extension and updated it to v3.

Current stage is _draft_ as some tweaks beyond what I've already done will need to be done.

Once completed, should officially fix issue #181.

### CODE REVIEW

**Comment by SpeedyD** [comment]
Requesting @Ne-wt, @itinerare and @AW0005 to look things over and comment where possible. Even though I jumped into this with all vigor, my brain is tired now. All ideas welcome!!

I'll also need to figure out how to make a migration that swaps the text from Characters to CharacterImages, possibly with an extra check..

And quite a bit of extra cleanup is necessary... I already did a few minor changes to make it all consistent, but I wouldn't be surprised if missed something obvious. ^^;;

Also: severely **_untested_** at this current point.

File: app/Models/User/User.php [hunk 1]
```diff
@@ -441,6 +441,24 @@ public function getBirthdayDisplayAttribute() {
...
+                return null;
+            break;
+            case 1:
+                return null;
+            break;
+            case 2:
+                return null;
+            break;
+        }
+    }
```
Comment (SpeedyD) [inline app/Models/User/User.php:461]: ...I have to actually wonder why this exists. I don't think it's used anywhere?

File: app/Services/UserService.php [hunk 2]
```diff
@@ -275,7 +275,22 @@ public function disableTwoFactor($code, $user) {
...
+     *
+     */
+    public function updateWarningVisibility($data, $user)
+    {
+        $user->settings->warning_visibility = $data;
+        $user->settings->save();
+
+        return true;
+    }
+
```
Comment (SpeedyD) [inline app/Services/UserService.php:291]: This WILL require additional tweaking. I'll probably change it to be similar to the birthday update.

**Review by SpeedyD — COMMENTED** [review]
These're two points I need to remember to edit soon. Comments welcome.

File: database/migrations/2023_10_12_011826_add_character_warning_to_character_profile.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('character_warning', function (Blueprint $table) {
```
Comment (Ne-wt) [inline database/migrations/2023_10_12_011826_add_character_warning_to_character_profile.php:33]: going to make a pr for some changes but this migration reverse is incorrect

Comment (SpeedyD) [reply database/migrations/2023_10_12_011826_add_character_warning_to_character_profile.php:33]: Oh shit, it is! Can you fix that in your PR as well?

File: app/Models/User/User.php (hunk 1 above)
Comment (SpeedyD) [reply app/Models/User/User.php]: Fixed by @redcr0wz.

File: database/migrations/2023_10_12_011826_add_character_warning_to_character_profile.php (hunk 3 above)
Comment (SpeedyD) [reply database/migrations/2023_10_12_011826_add_character_warning_to_character_profile.php]: Thanks for the notif, @redcr0wz fixed it.

**Comment by SpeedyD** [comment]
Nice work, @itinerare and @ScuffedNewt!..

Looks like there are a few conflicts to be resolved, lemme look into those quickly.

**Comment by itinerare** [comment]
Happy to help :sparkles:

**Comment by SpeedyD** [comment]
Is that change in [App/Services/UserService.php](https://github.com/corowne/lorekeeper/pull/876/files#diff-28a0b3aa7c8bcb21672fcfd472204061d19039d51ef8b0da41b268c070eae085) still required? I intended to give it a change similar to the birthday update back when, but.. as with everything regarding this specific PR, it fell off the grid =_=;

**Comment by ScuffedNewt** [comment]
> Is that change in [App/Services/UserService.php](https://github.com/corowne/lorekeeper/pull/876/files#diff-28a0b3aa7c8bcb21672fcfd472204061d19039d51ef8b0da41b268c070eae085) still required? I intended to give it a change similar to the birthday update back when, but.. as with everything regarding this specific PR, it fell off the grid =_=;

yup iirc! it's used on the settings page

**Comment by SpeedyD** [comment]
On suggestion of @ScuffedNewt, I'm putting this on ready for review. I'll resolve that BrowseController conflict in a moment.

**Comment by SpeedyD** [comment]
Technically left to do:
- The earlier comment regarding UserService
- Check out all the empty TODOs, they were added for some reason, I'd assume.

Nonetheless, please review as is. :)

**Comment by SpeedyD** [comment]
..helloyello, the pint action didn't run cause it's announcing
>  ! app/Models/Character/CharacterImage.php Parse error: syntax error, unexpe…  
..wish I knew the rest of that line.

**Comment by itinerare** [comment]
That means the file is malformed somehow-- looks like a function got its `}`(s) chopped off.

**Comment by SpeedyD** [comment]
Closing after discussion with @ScuffedNewt. Please review #1052 instead. :)

**Pull request closed without merging**